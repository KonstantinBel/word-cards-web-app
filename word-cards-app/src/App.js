// @flow

import './App.scss';
import React, { Component, Suspense } from 'react';
import { Route } from 'react-router-dom';
import cookie from 'js-cookie';
import AnotherPage from './components/example-app-page/example-app-page';
import RouteExample from './components/route-example/route-example';
import Menu from './components/menu';
import LanguageToggler from './components/language-toggler';
import getLocale from './locale';
import AppContext from './app-context';

const Auth = React.lazy(() => import('./components/auth/auth'));

class App extends Component {
  state = { locale: getLocale(cookie.get('userlang') || navigator.language || navigator.userLanguage) };

  setLocale = (localeName) => {
    cookie.set('userlang', localeName);
    this.setState({ locale: getLocale(localeName) });
  }

  render() {
    const { match } = this.props;
    const { locale } = this.state;

    return (
      <AppContext.Provider value={{ locale }}>
        <div>
          {process.env.REACT_APP_DEV && <span style={{ color: 'red' }}>starting in dev mode</span>}
          <h1>{locale.mainPage.header}</h1>
          <p>{locale.mainPage.mainContent}</p>
          <LanguageToggler clickHandler={this.setLocale} />
          <Route exact path={match.path} component={Menu} />
          {/* lazyload page example */}
          <Route
            path={`${match.path}/auth`}
            render={props => (
              <Suspense fallback={<div>Loading...</div>}>
                <Auth {...props} />
              </Suspense>
            )}
          />
          <Route path={`${match.path}/page`} component={AnotherPage} />
          <Route path={`${match.path}/route-example`} component={RouteExample} />
          <a href="/">Go to base page</a>
        </div>
      </AppContext.Provider>
    );
  }
}

export default App;
