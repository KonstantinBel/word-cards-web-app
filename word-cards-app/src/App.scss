@import './components/auth/auth.scss'
