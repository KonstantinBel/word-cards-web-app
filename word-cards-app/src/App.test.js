import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <App match={{ path: 'qwe' }} />
      </BrowserRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  test('есть корректный снимок', () => {
    const component = renderer.create(
      <BrowserRouter>
        <App match={{ path: 'qwe' }} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
