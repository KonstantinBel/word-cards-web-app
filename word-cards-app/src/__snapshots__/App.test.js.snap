// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App есть корректный снимок 1`] = `
<div>
  <h1>
    App main page title
  </h1>
  <p>
    App main page content
  </p>
  <div>
    <button
      onClick={[Function]}
    >
      RU
    </button>
    <button
      onClick={[Function]}
    >
      EN
    </button>
  </div>
  <a
    href="/"
  >
    Go to base page
  </a>
</div>
`;
